### ISSUE TITLE
Update Unicode character width determination to support newer versions of Unicode

### ISSUE THREAD

**Issue by meonkeys** [GothenburgBitFactory/taskwarrior#2333]
Emoji characters muck up the automatic terminal width detection and cause lines to wrap unexpectedly. Example:

![2020-10-17T12-13-20-screenshot](https://user-images.githubusercontent.com/50639/96351738-3f584e00-1072-11eb-90c3-213e29ecc866.png)

Notice how the final digit in the urgency value appears on the next line.

This screenshot is from taskwarrior v2.5.1 on Ubuntu 20.04.1 64-bit LTS desktop.

Terminal details:

```
$ gnome-terminal --version
# GNOME Terminal 3.36.2 using VTE 0.60.3 +BIDI +GNUTLS +ICU +SYSTEMD
```

I think the glyphs are from the `fonts-noto-color-emoji` package.

**Comment by tbabej** [comment]
Reproduced on 2.6.0: It looks like we are missing some Unicode characters can take 2 characters in the terminal when expected to only take 1:

![image](https://user-images.githubusercontent.com/2123868/100304679-488be300-2f6d-11eb-8681-206a1ce4e50d.png)

Using [`wcwidth` library](https://github.com/jquast/wcwidth) in Python we can see that this charater's printable length is 2:
```python
>>> from wcwidth import wcwidth
>>> wcwidth('👍')
2
```

Further investigation shows that this is due to the fact that my chosen character 👍' was defined in Unicode version 6.0, while our [current implementation of `mk_wcwidth`](https://github.com/GothenburgBitFactory/libshared/blob/master/src/wcwidth6.cpp) only supports up to version 5.0.

**Comment by tbabej** [comment]
Indeed, looks like for emojis, the defined length was 1 until Unicode 9.0. The following C++11 header-only library should be a good replacement for the one we're currently using: https://github.com/ridiculousfish/widecharwidth

**Comment by tbabej** [comment]
Fixed in 2.6.0.
